.github/workflows/release.yml

```diff
@@ -0,0 +1,51 @@
+name: Release Chart
+
+on:
+  push:
+    branches:
+      - main
+
+jobs:
+  release:
+    permissions:
+      contents: write
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - name: Configure Git
+        run: |
+          git config user.name "$GITHUB_ACTOR"
+          git config user.email "$GITHUB_ACTOR@users.noreply.github.com"
+
+      - name: Generate Chart
+        run: |
+          make helm-chart-generate
+
+      - name: Check for changes to generated helm chart
+        id: check_changes
+        run: |
+          git status --porcelain charts/ | grep -q "." && echo "changes=true" >> $GITHUB_OUTPUT || echo "changes=false" >> $GITHUB_OUTPUT
+
+      # If we generated a new helm chart, we need to commit it
+      # We need to do this step because our helm chart is not the source of truth, it's generated from starter
+      # Note that failing to do this doesn't break this release, but it can break future ones and make the release state confusing
+      - name: Commit generated chart
+        id: update_source
+        run: |
+          git pull
```

Comment:
is it necessary to do a pull here? it seems unnecessary to me given we already have the full history and the workflow runs on a push to main.

if you feel strongly about it, how about `git pull --rebase origin main` instead of just a plain pull?

Reply:
I think I ran into an issue with one of my experimental runs. Good call, I'll do a `git pull -r`